sycl/include/sycl/exception.hpp

```diff
@@ -53,11 +54,12 @@ __SYCL_EXPORT std::error_code make_error_code(sycl::errc E) noexcept;
 __SYCL_EXPORT const std::error_category &sycl_category() noexcept;
 
 namespace detail {
-__SYCL_EXPORT const char *stringifyErrorCode(pi_int32 error);
+__SYCL_EXPORT const char *stringifyErrorCode(int32_t error);
 
-inline std::string codeToString(pi_int32 code) {
-  return std::string(std::to_string(code) + " (" + stringifyErrorCode(code) +
-                     ")");
+inline std::string codeToString(int32_t code) {
+  std::stringstream ss;
+  ss << stringifyErrorCode(code);
```

Comment:
Why do we need `string stream` to convert `const char *` into `std::string`? That could be done just by `std::string` constructor.

I'm not familiar with `stringstream` implementation, but if it is better at reducing amount of temporary allocations, then we can consider using it for producing the whole string to avoid `string::operator+` usage that re-allocates and copies every time.

Reply:
Good point, changed that to `std::string` as it should be more efficient as I know.